File: SimulasiAPBN.Core/Models/Budget.cs
/*
 * Simulasi APBN
 *
 * Program ditulis oleh Danang Galuh Tegar Prasetyo (https://danang.id/)
 * untuk Kementerian Keuangan Republik Indonesia.
 */
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using SimulasiAPBN.Core.Enumerators;
using DapperTable = Dapper.Contrib.Extensions.TableAttribute;
using DapperWrite = Dapper.Contrib.Extensions.WriteAttribute;

namespace SimulasiAPBN.Core.Models
{
    // For Simulation & Policy
    [DapperTable("Budgets")]
    public sealed class Budget : GenericModel
    {

        [Required]
        public string Function { get; set; }

        [Required]
        public BudgetType Type { get; set; }
        
        [Required]
        public string Description { get; set; }

        [DapperWrite(false)]
        public ICollection<BudgetTarget> BudgetTargets { get; set; }

    }
}

File: SimulasiAPBN.Web/React/src/core/models/Budget.ts
/**
 * Simulasi APBN
 *
 * Program ditulis oleh Danang Galuh Tegar Prasetyo (https://danang.id/)
 * untuk Kementerian Keuangan Republik Indonesia.
 */
import GenericModel from "@core/models/GenericModel"
import BudgetTarget from "@core/models/BudgetTarget"
import BudgetType from "@core/enums/BudgetType"

interface Budget extends GenericModel {
	function: string
	type: BudgetType
	description: string
	budget_targets: BudgetTarget[]
}

export default Budget
